### ISSUE TITLE
[Feature Request] [stdlib] `bit` module needs to be completed

### ISSUE THREAD

**Issue by LJ-9801** [modularml/mojo#2862]
### Review Mojo's priorities

- [X] I have read the [roadmap and priorities](https://docs.modular.com/mojo/roadmap.html#overall-priorities) and I believe this request falls within the priorities.

### What is your request?

There are several things that need to be done for the `bit` module after #2859 is merged. First of all, `bit_reverse`, `byte_reverse` and `pop_count` all needs to be implemented for type `Int`. These are probably just simple wrapper around LLVM IR. After `pop_count` is implemented, we can refactor out both implementation of `has_single_bit` as mentioned in the discussion in #2859. Second of all, we need to complete all the tests cases for the rest of the `bit` module.

### What is your motivation for this change?

complete entire `bit` module.

### Any other details?

*No response*

```[tasklist]
### Tasks
- [x] implement `bit_reverse`, `byte_swap`, `pop_count` in Int type
- [ ] use `popcount() == 1` to determine `is_power_of_two`
```

**Comment by JoeLoser** [comment]
Can we create sub-issues here for the remaining work (if any) so we know what remains?  That will make it easy for other contributors too.  Thanks!<br><br>CC: @laszlo too

**Comment by LJ-9801** [comment]
@JoeLoser just updated the task list. @laszlokindrat is there any other thing that needs to be done?

**Comment by laszlokindrat** [comment]
> @laszlokindrat is there any other thing that needs to be done?

Not that I can think of right now. Thanks for working on this~

**Comment by laszlokindrat** [comment]
@LJ-9801 Is the rationale for updating `is_power_of_two` to use `pop_count` just performance? If so, is there anything blocking it?

**Comment by LJ-9801** [comment]
> @LJ-9801 Is the rationale for updating `is_power_of_two` to use `pop_count` just performance? If so, is there anything blocking it?

@laszlokindrat The rationale for this change is mainly for encouraging reusing existing routine instead of redefining the same expression. Not sure if it will provide extra performance just from eye tests. I did tried to implement this in #3150 but it caused the 'bit' module test to fail for some reason. I can give it a go in another PR and try to figure out what is wrong.

**Comment by laszlokindrat** [comment]
Since `pop_count` uses an llvm intrinsic, it is not yet possible to use it in parameter contexts (i.e. at compile time). This might be the reason why you saw failures. Unless there is a perf difference you can demonstrate, I would suggest not changing the implementation.

**Comment by laszlokindrat** [comment]
This is complete now, since we will not change the implementation of `is_power_of_two` for now.
